### PR TITLE
Refresh multi-part WCF tutorial

### PR DESCRIPTION
**Pull request by gewarren**
- update VS references
- add WCF component info for VS
- add dev_langs metadata and code snippet language tags
- add Next steps buttons at the end of each task in the tutorial
- whitespace cleanup

Part of the fix for #7172.

### CODE REVIEW

File: docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md [hunk 1]
```diff
@@ -1,425 +1,433 @@
...
+2. Add a reference to the GettingStartedLib project to the GettingStartedHost project. Right-click on the **References** folder under the GettingStartedHost project in **Solution Explorer**, and then select **Add Reference**. In the **Add Reference** dialog, select **Solution** on the left-hand side of the dialog, select GettingStartedLib in the center section of the dialog, and then choose **Add**. This makes the types defined in GettingStartedLib available to the GettingStartedHost project.
+
+3. Add a reference to System.ServiceModel to the GettingStartedHost project. Right-click the **References** folder under the GettingStartedHost project in **Solution Explorer** and select **Add Reference**. In the **Add Reference** dialog, select **Framework** on the left-hand side of the dialog under **Assemblies**. Find and select **System.ServiceModel**, and then choose **OK**. Save the solution by selecting **File** > **Save All**.
+
+## Host the service
+
+Open the Program.cs or Module.vb file and enter the following code:
+
+```csharp
+// program.cs
```
Comment (rpetrusha) [inline docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md]: Here again, it's useful to pare down the namespaces to the ones actually used (the same ones as in the VB example, along with System. And the filename can be removed.

File: docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md [hunk 2]
```diff
@@ -1,425 +1,433 @@
...
+            {
+                Console.WriteLine("An exception occurred: {0}", ce.Message);
+                selfHost.Abort();
+            }
+        }
+    }
+}
+```
+
+```vb
```
Comment (rpetrusha) [inline docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md]: The filename can be removed, as can Imports System.

File: docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md [hunk 3]
```diff
@@ -1,425 +1,433 @@
...
+                Console.WriteLine("An exception occurred: {0}", ce.Message)
+                selfHost.Abort()
+            End Try
+        End Sub
+    End Class
+
+End Module
+```
+
+1. Step 1 - Creates an instance of the Uri class to hold the base address of the service. Services are identified by a URL which contains a base address and an optional URI. The base address is formatted as follows:[transport]://[machine-name or domain][:optional port #]/[optional URI segment]The base address for the calculator service uses the HTTP transport, localhost, port 8000, and the URI segment "GettingStarted"
```
Comment (rpetrusha) [inline docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md]: The use of a number and "Step x" seems redundant here.

File: docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md [hunk 4]
```diff
@@ -1,425 +1,433 @@
...
-            [OperationContract]  
-            double Multiply(double n1, double n2);  
-            [OperationContract]  
-            double Divide(double n1, double n2);  
-        }  
-}  
-```  
-  
+// IService1.cs
+using System;
```
Comment (rpetrusha) [inline docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md]: Pare down namespaces here.

File: docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md [hunk 5]
```diff
@@ -1,425 +1,433 @@
...
-            Console.WriteLine("Received Divide({0},{1})", n1, n2);  
-            Console.WriteLine("Return: {0}", result);  
-            return result;  
-        }  
-    }  
-}  
-```  
-  
+// Service1.cs
+using System;
```
Comment (rpetrusha) [inline docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md]: Pare down namespaces here.

File: docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md [hunk 6]
```diff
@@ -1,425 +1,433 @@
...
-                Console.WriteLine("An exception occurred: {0}", ce.Message);  
-                selfHost.Abort();  
-            }  
-        }  
-    }  
-}  
-```  
-  
+//Program.cs
+using System;
```
Comment (rpetrusha) [inline docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md]: Similarly, here.

File: docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md [hunk 7]
```diff
@@ -1,425 +1,433 @@
...
-        <OperationContract()> _  
-        Function Multiply(ByVal n1 As Double, ByVal n2 As Double) As Double  
-        <OperationContract()> _  
-        Function Divide(ByVal n1 As Double, ByVal n2 As Double) As Double  
-    End Interface  
-End Namespace  
-```  
-  
+'IService1.vb
+Imports System
```
Comment (rpetrusha) [inline docs/framework/wcf/how-to-host-and-run-a-basic-wcf-service.md]: Remove Imports System here and in the later VB code examples in this page.

File: docs/framework/wcf/how-to-implement-a-wcf-contract.md [hunk 8]
```diff
@@ -1,267 +1,278 @@
...
-            double Subtract(double n1, double n2);  
-            [OperationContract]  
-            double Multiply(double n1, double n2);  
-            [OperationContract]  
-            double Divide(double n1, double n2);  
-        }  
-}  
-```  
-  
+//Service1.cs
```
Comment (rpetrusha) [inline docs/framework/wcf/how-to-implement-a-wcf-contract.md]: The same comments as previously apply to both the VB and C# code here.

File: docs/framework/wcf/how-to-use-a-wcf-client.md [hunk 9]
```diff
@@ -1,143 +1,150 @@
...
+
+2.  Calls the service operations from the generated proxy.
+
+3.  Closes the client once the operation call is completed.
+
+## Use a Windows Communication Foundation client
+
+Open the Program.cs or Program.vb file from the GettingStartedClient project and replace the existing code with the following code:
+
+```csharp
```
Comment (rpetrusha) [inline docs/framework/wcf/how-to-use-a-wcf-client.md]: The same comments as earlier apply to VB and C# code in this article.

File: docs/framework/wcf/how-to-use-a-wcf-client.md [hunk 10]
```diff
@@ -1,143 +1,150 @@
...
+        Console.WriteLine()
+        Console.WriteLine("Press <ENTER> to terminate client.")
+        Console.ReadLine()
+
+    End Sub
+
+End Module
+```
+
+Notice the using or imports statement that imports the GettingStartedClient.ServiceReference1. This imports the code generated by Add Service Reference in Visual Studio. The code instantiates the WCF proxy and then calls each of the service operations exposed by the calculator service, closes the proxy and terminates.
```
Comment (rpetrusha) [inline docs/framework/wcf/how-to-use-a-wcf-client.md]: using, imports, and GettingStartedClientServiceReference1 can all be fenced.
Add Service Reference can be bolded.
comma after "closed the proxy"

File: docs/framework/wcf/wcf-service-publishing.md [hunk 11]
```diff
@@ -4,67 +4,67 @@ ms.date: "03/30/2017"
...
+
+## Supported Services and Target Locations
+ WCF Service Publishing supports publishing WCF services created from the set of WCF service library templates, and their corresponding item templates, which include the following:
+
+-   WCF Service Library template with item template.
+
+-   Syndication Service Library.
+
+ You can find these service templates by choosing **File** -> **New Project** -> **Visual Basic** or **Visual C#** -> **WCF**. For other WCF templates in this location (including WCF Workflow Service Application and WCF Service Application) you can publish using [One-Click publishing for web applications](https://msdn.microsoft.com/library/dd465337\(v=vs.110\).aspx).
```
Comment (rpetrusha) [inline docs/framework/wcf/wcf-service-publishing.md]: nit: comma before "you can publish"

File: docs/framework/wcf/wcf-service-publishing.md [hunk 12]
```diff
@@ -4,67 +4,67 @@ ms.date: "03/30/2017"
...
+
+     If you cannot browse to the site, it may because directory browser is not enabled in IIS. Please follow the tips in the "Things you can try" section to enable it. Alternatively, you can directly type" HYPERLINK "http://localhost/WebApplicationFolderName" http://localhost/WebApplicationFolderName/ProjectName.ServiceName.svc" to view your service page.
+
+ You can use **Publish** to specify if you want to copy the assembly, configuration, and .svc file for all services defined in the project to the target location, and overwrite existing files at the destination.
+
+ If you choose to deploy your application to local IIS, you may encounter errors related to IIS setup. Please ensure that IIS is installed properly. You can type " HYPERLINK "http://localhost" http://localhost" in your browser and check whether the IIS default page is showing up.  In some cases, the issues may also be caused by ASP.NET or WCF improper registration in IIS. You can open the Visual Studio Command Prompt and run the command "aspnet_regiis.exe -ir" to fix ASP.NET registration issues, or run command "ServiceModelReg.exe –ia" to fix WCF registration issues.
```
Comment (rpetrusha) [inline docs/framework/wcf/wcf-service-publishing.md]: " HYPERLINK "http://localhost" http://localhost" --> \`http://localhost\`

**Review by rpetrusha — CHANGES_REQUESTED** [review]
You've substantially improved these topics, @gewarren. I've left a number of comments, many of them unrelated to your PR but that it would be nice to fix. Also, this is a substantial enough revision that you should update the ms.date field.

**Comment by gewarren** [comment]
@rpetrusha Thanks for reviewing this!

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
Thanks, @gewarren, for making the additional changes. You can merge whenever you're ready.

**Comment by gewarren** [comment]
@rpetrusha Thank you!

**Pull request merged**